app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt

```diff
@@ -0,0 +1,122 @@
+package org.jellyfin.mobile.player
+
+import android.app.Application
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build
+import androidx.core.content.getSystemService
+import org.jellyfin.mobile.AppPreferences
+import org.jellyfin.mobile.BuildConfig
+import org.jellyfin.mobile.R
+import org.jellyfin.mobile.utils.Constants
+import org.jellyfin.mobile.utils.Constants.VIDEO_PLAYER_NOTIFICATION_ID
+import org.jellyfin.mobile.utils.createMediaNotificationChannel
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+import java.util.concurrent.atomic.AtomicBoolean
+
+class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinComponent {
+    private val appPreferences: AppPreferences by inject()
+    private val context: Context = viewModel.getApplication<Application>()
+    private val notificationManager: NotificationManager? by lazy { context.getSystemService() }
+    private val receiverRegistered = AtomicBoolean(false)
+
+    val shouldShowNotification: Boolean
+        get() = appPreferences.exoPlayerAllowBackgroundAudio
+
+    @Suppress("DEPRECATION")
+    fun postNotification() {
+        if (!shouldShowNotification) return
+        val nm = notificationManager ?: return
+        val mediaSource = viewModel.mediaSourceManager.jellyfinMediaSource.value ?: return
+        val player = viewModel.playerOrNull ?: return
+
+        // Create notification channel
+        context.createMediaNotificationChannel(nm)
+
+        val style = Notification.MediaStyle().apply {
+            setMediaSession(viewModel.mediaSession.sessionToken)
+            //setShowActionsInCompactView(0, 1, 2) // TODO
```

Comment:
`// TODO`

Reply:
Yeah, still mergeable though. For some reason, if you set that value, the notification always stays in compact mode, which also kills the seek bar. I suppose that happens because we don't have a cover art, but I want/need to pull in the api client for that first. That'll happen later.

Reply:
Maybe add some additional information in that case;
```suggestion
            // FIXME Doesn't work properly until cover art is added
            //setShowActionsInCompactView(0, 1, 2)
```

something like this,